### PR TITLE
Only retrieve file info when that's all we need, in the Browse view

### LINKED ISSUE #1590: Only retrieve file info when that's all we need, in the Browse view
This is a spin-off from #1574 

While working on that issue I found a number of things that can/should be fixed in terms of how we're calling APIs and returning data for both the Browse and Compare views. Because this is a fairly large task, I'm breaking it down into smaller pieces.

This issue is simply about changing the logic for the Browse view so that when we only need to retrieve file information (because we already have version information), we use the new `file_only` param of the browse API to avoid returning unnecessary data.

For QA:

When looking at the API requests generated by code manager to `/api/v5/reviewers/addon/{addonId}/versions/{versionId}/`, the first API request for a version will *not* have the `file_only` param specified, and the response will be a complete `version` object, including a `file_entries` property and an embedded complete `file` object. For each subsequent request to the API for a different file (there will be a `file=` param on the URL), bur for the same version, the `file_only=true` param should be specified, and the response will contain just an `id` and a `file` object.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #1590

### CODE REVIEW

File: src/api/index.tsx [hunk 1]
```diff
@@ -302,6 +306,17 @@ export const fetchAllPages = async <ResultsType,>(
   );
 };
 
+const makeApiQuery = (path?: string, fileOnly?: boolean) => {
+  const query: CallApiQuery = {};
+  if (path) {
```
Comment (eviljeff) [inline src/api/index.tsx:311]: when would `path` ever be missing/empty?

File: src/reducers/versions.tsx [hunk 2]
```diff
@@ -1333,6 +1361,27 @@ export const createReducer = ({
...
         };
       }
+      case getType(actions.loadVersionFileFromVersion): {
+        const { path, version } = action.payload;
+
+        return {
+          ...state,
+          versionFiles: {
+            ...state.versionFiles,
+            [version.id]: {
```
Comment (eviljeff) [inline src/reducers/versions.tsx:1371]: I'm reading this (perhaps through my python sunglasses) as wrapping `version.id` in a list and making the list an object key/property ... which seems a strange thing to do.

File: src/api/index.tsx (hunk 1 above)
Comment (bobsilverberg) [reply src/api/index.tsx:311]: When you go from Reviewer Tools to Code Manager it takes you to the default file, which is generally `manifest.json`, but at that point no file is actually selected so no `path` is sent to the API.

File: src/reducers/versions.tsx (hunk 2 above)
Comment (bobsilverberg) [reply src/reducers/versions.tsx:1371]: It makes sense that your python sunglasses would show you that. If you switch those for these JS sunglasses 🕶️ , then you'll see that what's being said here is that the name of the property in question is the value of `version.id`, so if `version.id == 1` then the property would be called `1`.

Comment (eviljeff) [reply src/reducers/versions.tsx:1371]: interesting.  I guess the obvious ` version.id: {` wouldn't work because Javascript's syntax is so overloaded.

Comment (bobsilverberg) [reply src/reducers/versions.tsx:1371]: Yeah, I'm pretty sure that's not legal.

**Pull request merged**